### DISCUSSION TITLE
[10 gennaio 2024] Automi

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/automi-calcolabilita-complessita#19] [Automi]
![image](https://github.com/sapienzastudentsnetwork/acc2324/assets/69389461/199ce797-45da-4de4-b705-b79ad5a709d5)

**Comment by Exyss** [comment]
Per il primo punto, ripropongo le seguenti due soluzioni presenti nei miei appunti:

![image](https://github.com/sapienzastudentsnetwork/acc2324/assets/65043207/f8e750bd-4768-4584-a70c-c59da6514dde)

----

La dimostrazione richiesta nel secondo punto è presente sul libro di testo

L'esempio di applicazione coincide con quello dell'esercizio #7

**Reply by Loriv3** [reply]
Ciao, non riesco a capire bene la costruzione del NFA. In questo caso gli $\varepsilon$-archi non dovrebbero avere anche le altre lettere dell'alfabeto? Spiegandomi meglio, gli archi q3 -> q1 e q4 -> q1 non dovrebbero avere come etichette \{ 1, $\varepsilon$ \}?

**Reply by Exyss** [reply]
Gli $\varepsilon$-archi vengono automaticamente percorsi durante la computazione, senza leggere nulla dall'input

**Reply by dahKidou** [reply]
Ma se il NFA si trova a calcolare l' input w = "0" cosa fa? Essendo q1 lo stato iniziale, accettante, leggendo 0 dovrebbe restare in q1 e quindi accettare oppure devo farmi un ripasso di teoria io?

**Reply by Exyss** [reply]
Visto che non ci sono archi uscenti segnati con 0, la computazione muore appena legge 0. Quando una transizione non è definita per un NFA l'intero ramo di computazione muore.

**Comment by AglaiaNorza** [comment]
![1](https://github.com/user-attachments/assets/fbd71c99-a433-4169-91fb-1e310f3472d9)

**Comment by daromacarlo** [comment]
Potrebbe valere come soluzione?

<img width="1716" height="600" alt="immagine" src="https://github.com/user-attachments/assets/b2fafcb6-d4b4-45a8-bf03-b82c17466bf4" />

**Reply by daromacarlo** [reply]
<img width="1719" height="325" alt="immagine" src="https://github.com/user-attachments/assets/bba56584-1a45-4399-b7ad-6b167ee7d3bb" />
